packages/api/src/controllers/room.ts

```diff
@@ -64,14 +72,163 @@ function toExternalRoom(room: Room, isAdmin = false) {
+  }
+
+  const svc = new RoomServiceClient(
+    req.config.livekitHost,
+    req.config.livekitApiKey,
+    req.config.livekitSecret
+  );
+  await svc.deleteRoom(req.params.roomId);
+
+  room.deleted = true;
+  room.deletedAt = new Date().getTime();
+  await db.room.replace(room);
+
+  res.status(204).end();
+});
+
+app.post(
+  "/:roomId/egress",
+  authorizer({}),
+  validatePost("room-egress-payload"),
+  async (req, res) => {
+    const room = await db.room.get(req.params.roomId);
+
+    if (!req.user.admin && req.user.id !== room.userId) {
+      throw new ForbiddenError(`users may only modify their own rooms`);
+    }
+
+    if (room.egressId !== undefined) {
+      throw new BadRequestError("egress already started");
+    }
+
+    const egressClient = new EgressClient(
+      req.config.livekitHost,
+      req.config.livekitApiKey,
+      req.config.livekitSecret
+    );
+    const output: StreamOutput = {
+      protocol: StreamProtocol.RTMP,
+      urls: [req.body.rtmpURL],
+    };
+
+    const info = await egressClient.startRoomCompositeEgress(
+      req.params.roomId,
+      output,
+      {
+        layout: "speaker-dark",
+        encodingOptions: {
+          keyFrameInterval: 2,
+        },
+      }
+    );
+    console.log("egress started", info);
+    room.egressId = info.egressId;
+    room.updatedAt = new Date().getTime();
+    await db.room.replace(room);
+    res.status(204).end();
+  }
+);
+
+app.delete("/:roomId/egress", authorizer({}), async (req, res) => {
+  const room = await db.room.get(req.params.roomId);
+
+  if (!req.user.admin && req.user.id !== room.userId) {
+    throw new ForbiddenError(`users may only modify their own rooms`);
+  }
+
+  if (room.egressId === undefined) {
+    throw new BadRequestError("egress has not started");
+  }
+
+  const egressClient = new EgressClient(
+    req.config.livekitHost,
+    req.config.livekitApiKey,
+    req.config.livekitSecret
+  );
+
+  const info = await egressClient.stopEgress(room.egressId);
+  console.log("egress stopped", info);
+  room.egressId = undefined;
+  room.updatedAt = new Date().getTime();
+  await db.room.replace(room);
+  res.status(204).end();
+});
+
+app.post(
+  "/:roomId/user",
+  authorizer({}),
+  validatePost("room-user-payload"),
+  async (req, res) => {
+    const room = await db.room.get(req.params.roomId);
+
+    if (!req.user.admin && req.user.id !== room.userId) {
+      throw new ForbiddenError(`users may only add users to their own rooms`);
+    }
+
+    const id = uuid();
+    const at = new AccessToken(
+      req.config.livekitApiKey,
+      req.config.livekitSecret,
+      {
+        name: req.body.name,
+        identity: id,
+        ttl: 5 * 60,
+      }
+    );
+    at.addGrant({ roomJoin: true, room: req.params.roomId });
+    const token = at.toJwt();
+
+    res.status(201);
+    res.json({
+      id: id,
+      joinUrl:
+        "https://meet.livekit.io/custom?liveKitUrl=" +
```

Comment:
Are we hosting a custom version of this ourselves at some point? Might wanna make this domain configurable already if so.